test-suite/js/src/test/require-sam/org/scalajs/testsuite/jsinterop/CustomJSFunctionTest.scala

```diff
@@ -0,0 +1,112 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2018, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.testsuite.jsinterop
+
+import scala.scalajs.js
+
+import org.junit.Assert._
+import org.junit.Test
+
+import org.scalajs.testsuite.utils.JSAssert._
+
+class CustomJSFunctionTest {
+
+  import CustomJSFunctionTest._
+
+  @Test def customJSFunctions(): Unit = {
+    val array = js.Array(4, 6, 2, 3).asInstanceOf[js.Dynamic]
+
+    val getAllArgs = new js.Function("...args", "return args;")
```

Comment:
Consider putting this and the related tests into a different test case: It tests the calling convention, and not the implementation of the functions. (TBH it feels like we should have tests of this already, this is nothing new. I haven't checked though).